optaplanner-examples/src/main/java/org/optaplanner/examples/common/experimental/impl/ConsecutiveIntervalInfoImpl.java

```diff
@@ -0,0 +1,282 @@
+/*
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.examples.common.experimental.impl;
+
+import java.util.NavigableMap;
+import java.util.NavigableSet;
+import java.util.TreeMap;
+import java.util.TreeSet;
+import java.util.function.BiFunction;
+
+import org.apache.commons.lang3.ObjectUtils;
+import org.optaplanner.examples.common.experimental.api.ConsecutiveIntervalInfo;
+import org.optaplanner.examples.common.experimental.api.IntervalBreak;
+import org.optaplanner.examples.common.experimental.api.IntervalCluster;
+
+public class ConsecutiveIntervalInfoImpl<Interval_, Point_ extends Comparable<Point_>, Difference_ extends Comparable<Difference_>>
+        implements
+        ConsecutiveIntervalInfo<Interval_, Point_, Difference_> {
+    private final NavigableMap<IntervalSplitPoint<Interval_, Point_>, IntervalClusterImpl<Interval_, Point_, Difference_>> clusterStartSplitPointToCluster;
+    private final NavigableSet<IntervalSplitPoint<Interval_, Point_>> splitPointSet;
+    private final NavigableMap<IntervalSplitPoint<Interval_, Point_>, IntervalBreakImpl<Interval_, Point_, Difference_>> clusterStartSplitPointToNextBreak;
+    private final Iterable<IntervalCluster<Interval_, Point_, Difference_>> intervalClusterIterable;
+    private final BiFunction<Point_, Point_, Difference_> differenceFunction;
+    private final Iterable<IntervalBreak<Interval_, Point_, Difference_>> breaksIterable;
+
+    public ConsecutiveIntervalInfoImpl(TreeSet<IntervalSplitPoint<Interval_, Point_>> splitPointSet,
+            BiFunction<Point_, Point_, Difference_> differenceFunction) {
+        clusterStartSplitPointToCluster = new TreeMap<>();
+        clusterStartSplitPointToNextBreak = new TreeMap<>();
+        intervalClusterIterable = new MapValuesIterable<>(clusterStartSplitPointToCluster);
+        breaksIterable = new MapValuesIterable<>(clusterStartSplitPointToNextBreak);
```

Comment:
For the sake of consistency, I'd use `this` in the whole method, not just where it is necessary below.